test/trip_dispatch_test.rb

```diff
@@ -122,4 +119,49 @@ def build_test_dispatcher
       end
     end
   end
-end
+
+  describe "request_trip" do
+    before do
+      @dispatcher = build_test_dispatcher
+      @passenger_id = 3
+      @trip = @dispatcher.request_trip(@passenger_id)
+    end
+
+    it "creates a new trip" do
+      expect(@trip).must_be_kind_of RideShare::Trip
+      expect(@trip.id).must_be_instance_of Integer
+      expect(@trip.driver_id).must_equal 2
+      expect(@trip.start_time).must_be_instance_of Time
+      expect(@trip.end_time).must_be_nil
+      expect(@trip.cost).must_be_nil
+      expect(@trip.rating).must_be_nil
+    end
+
+    it "updates the driver status" do
+      expect(@trip.driver.status).must_equal :UNAVAILABLE
+    end
+
+    it "adds trip to the driver's trips" do
+      expect(@trip.driver.trips).must_include @trip
+    end
+  
+    it "adds trip to the passenger's trips" do
+      expect(@trip.passenger.trips).must_include @trip
+    end
+
+    it "adds trip to total trips list" do
+      expect(@dispatcher.trips).must_include @trip
+    end
+
+    it "returns a new trip" do
+      expect(@trip).must_equal @trip
+    end
+
+    it "raises ArgumentError if all drivers are unavailable" do
+      @dispatcher.drivers.each do |driver|
+           driver.status = :UNAVAILABLE
+      end
+      expect{@dispatcher.request_trip(3)}.must_raise ArgumentError
```

Comment:
Adding some spacing can make this read a little clearer.

```suggestion
      expect { @dispatcher.request_trip(3) }.must_raise ArgumentError
```